bots/psuteam7bot/castle.js

```diff
@@ -109,23 +128,29 @@ castle.recordPosition = (self) => {
  * Output: positions of other friendly castles.
  */
 castle.findPosition = (self) => {
-    const bots = self.getVisibleRobotMap().filter(bots =>{
-        return bots.team === self.me.team && bots.units === 0;
+    const bots = self.getVisibleRobots().filter(bots =>{
+        return bots.team === self.me.team && bots.castle_talk;
```

Comment:
Think this might need to be `castle_talk >= 0`? Could be wrong, but think it's `-1` if no signal so this might get things that aren't using `castle_talk`.

Reply:
There is a check for the `castle_talk` value to be between `[0, 64)` which was planned to be reserved for units on higher turns, but we can change that to `>= 0` to avoid the `-1`